test/ruby/test_compile_prism.rb

```diff
@@ -488,30 +488,8 @@ def test_PinnedVariableNode
     ############################################################################
 
     def test_ScopeNode
-      assert_separately(%w[], "#{<<-'begin;'}\n#{<<-'end;'}")
-                        begin;
-    def compare_eval(source)
-      ruby_eval = RubyVM::InstructionSequence.compile(source).eval
-      prism_eval = RubyVM::InstructionSequence.compile_prism(source).eval
-
-      assert_equal ruby_eval, prism_eval
-    end
-
-    def test_prism_eval(source)
-      $VERBOSE, verbose_bak = nil, $VERBOSE
-
-      begin
-        compare_eval(source)
-
-        # Test "popped" functionality
-        compare_eval("#{source}; 1")
-      ensure
-        $VERBOSE = verbose_bak
-      end
-    end
-        test_prism_eval("a = 1; tap do; { a: }; end")
-        test_prism_eval("a = 1; def foo(a); a; end")
-                        end;
+      test_prism_eval("a = 1; tap do; { a: }; end")
+      test_prism_eval("a = 1; def prism_test_scope_node(a); a; end")
```

Comment:
Doesn’t this define the global method?

I tried a singleton method but this still seems to define it in the top-level.

```suggestion
      test_prism_eval("a = Object; def a.prism_test_scope_node(a); a; end")
```